### DISCUSSION TITLE
[Idea] Add the possibility to call BO javascript events in our modules like in FO

### DISCUSSION THREAD

**Discussion by RomsC** [PrestaShop/PrestaShop#34599] [Ideas]
### The problem

Hi,

I'm developing a module and for one of my needs I have to add some JS in the order creation page.
I need to be able to execute JS each time the cart is loaded. There's an "OrderCreateCartLoaded" event defined in an EventEmitter and used for this purpose.
However, I don't have access to this EventEmitter from my module.

### The solution

Use a single EventEmitter for all the BO like in the FO with prestashop.on and prestashop.emit

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @RomsC 

Have you tried to use: `prestashop.component.EventEmitter`?

**Reply by mcaldex** [reply]
Hi @kpodemski ,

from my own testing, the `prestashop.component.EventEmitter` instance is available only after a while, it's not available straight away.
Sure, it's available within the browser console after the page loads, but not within my very own javascript code which gets executed before the prestashop components like the EventEmitter one.

My issue is that I'm looking to instantiate Grid javascript functionalities by leveraging the global 'prestashop' object, which I could only do after these components are loaded by the prestashop core.

In other words, I'd need to load my extension within this callback:
`EventEmitter.on('PSComponentsInitiated', () => {
  [...]
})`

but I do not have access to EventEmitter, because it's treated like a component like the grid ones.

**Reply by shahbaz-mehar** [reply]
Facing same issue
